src/views/catalog/templatescatalog/components/TemplatesCatalogDrawer/StorageSection/CustomizeSource/StorageClassSelect/StorageClassSelect.tsx

```diff
@@ -0,0 +1,55 @@
+import React, { FC, MouseEvent, useMemo, useState } from 'react';
+
+import useDefaultStorageClass from '@kubevirt-utils/hooks/useDefaultStorage/useDefaultStorageClass';
+import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+import { FormGroup, Select, SelectOption } from '@patternfly/react-core';
+
+type StorageClassSelectProps = {
+  onStorageClassChange: (_: MouseEvent, value: string) => void;
+  storageClassName: string;
+  storageClassRequired: boolean;
+};
+
+const StorageClassSelect: FC<StorageClassSelectProps> = ({
+  onStorageClassChange,
+  storageClassName,
+  storageClassRequired,
+}) => {
+  const { t } = useKubevirtTranslation();
+  const [{ storageClasses }] = useDefaultStorageClass();
+
+  const [isOpenStorageClass, setIsOpenStorageClass] = useState<boolean>(false);
+
+  const sortedStorageClasses = useMemo(
+    () => storageClasses?.map((sc) => sc?.metadata?.name)?.sort(),
+    [storageClasses],
+  );
+  if (!storageClassRequired) return null;
+
+  return (
+    <FormGroup
+      className="disk-source-form-group select-source-option"
+      isRequired
+      label={t('Storage class')}
+    >
+      <Select
+        onSelect={(event: MouseEvent, value: string) => {
+          onStorageClassChange(event, value);
+          setIsOpenStorageClass(false);
+        }}
+        hasInlineFilter
+        isOpen={isOpenStorageClass}
+        maxHeight={200}
+        onToggle={setIsOpenStorageClass}
+        placeholderText={t('Select storage class')}
+        selections={storageClassName}
+      >
+        {sortedStorageClasses?.map((name) => (
+          <SelectOption key={name} value={name} />
+        ))}
+      </Select>
+    </FormGroup>
```

Comment:
maybe I'm missing but where do u set the value of the dropdown to template default and, if not exist, to virt default and, if not exist, to cluster default, and if not exist, to first find?